### PR TITLE
NEB calculator with example of LiFePO4 by M3GNet-MS, M3GNet-DIRECT, and CHGNet

### PR DESCRIPTION
**Pull request by JiQi535**
## Summary
Added NEBCalc for NEB calculations with universal potentials. LiFePO4 is used as an example system, where M3GNet-MS, M3GNet-DIRECT, and CHGNet provide reasonable agreements to each other and to literature.

Major changes:

- Added a NEBCalc to perform NEB calculations with ASE.
- Included examples of path along b and c directions in LiFePO4 with M3GNet-MS, M3GNet-DIRECT, and CHGNet.
- NEBCalc can be constructed with from_end_images method that utilize Structure.interpolate in pymatgen to create initial NEB images.

## Checklist

- [x] Google format doc strings added. Check with `ruff`.
- [x] Type annotations included. Check with `mypy`.
- [x] Tests added for new features/fixes.
- [ ] If applicable, new classes/functions/modules have [`duecredit`](https://github.com/duecredit/duecredit) `@due.dcite` decorators to reference relevant papers by DOI ([example](https://github.com/materialsproject/pymatgen/blob/91dbe6ee9ed01d781a9388bf147648e20c6d58e0/pymatgen/core/lattice.py#L1168-L1172))

### CODE REVIEW

File: examples/NEB_data/traj_b_CHGNet/image_0.traj
Comment (janosh) [inline examples/NEB_data/traj_b_CHGNet/image_0.traj:1]: I don't see you loading all these trajectory files in the test nor example notebook you added. Were they committed by accident?

**Review by janosh — CHANGES_REQUESTED** [review]
Nice work @JiQi535, this looks great! 👍 

I think we need to cut down on the amount of `NEB_data` before merging as there are several MB of (unused?) traj data in this PR.

File: examples/NEB_data/traj_b_CHGNet/image_0.traj
Comment (JiQi535) [reply examples/NEB_data/traj_b_CHGNet/image_0.traj:1]: These can be deleted. They are truly not used anywhere. I will remove them from the PR.

Comment (JiQi535) [reply examples/NEB_data/traj_b_CHGNet/image_0.traj:1]: I have removed the traj files, and users can reproduce them with the notebook if needed. Now the NEB_data is below 1MB that majorly comes from the two png files in the notebook. Let me know if we need to futher downsize it or not.

File: matcalc/neb.py [hunk 1]
```diff
@@ -0,0 +1,137 @@
...
+                Default to 0.1 eV/Angstrom.
+            max_steps (int): Maximum number of steps in NEB calculations. Default to 1000.
+
+        Returns:
+            NEB barrier.
+        """
+        if self.traj_folder is not None:
+            os.makedirs(self.traj_folder, exist_ok=True)
+            for i in range(len(self.images)):
+                self.optimizer.attach(
```
Comment (janosh) [inline matcalc/neb.py:131]: We need some more tests for uncovered lines (see the codecov comments).

Screenshot 2023-12-11 at 9 53 38 AM

File: examples/NEB_data/traj_b_CHGNet/image_0.traj
Comment (shyuep) [reply examples/NEB_data/traj_b_CHGNet/image_0.traj:1]: Just use the PNG. Don't be silly. there are instances where vector formats are better and there are those where non-vector are better. this is one instance. Though 2.5 Mb is really nothing.

Comment (janosh) [reply examples/NEB_data/traj_b_CHGNet/image_0.traj:1]: @JiQi535 I think keeping visuals is nice. I compressed the PNG to 15% of orig size with minimal quality loss. We're good here!

**Comment by janosh** [comment]
Thanks @JiQi535, really nice contribution!

**Comment by JiQi535** [comment]
> Thanks @JiQi535, really nice contribution!

@janosh My pleasure to make some contributions here. And thanks for the thorough inspections and improvements before merging it!

**Pull request merged**